Use OnBecomingIdle to trigger idle actions on aircraft

reaperrr opened the pull request:
Like circling or landing.
Also added `IdleTurnSpeed` to allow to set a different turn speed for idle circling and in turn removed hardcoded 1/3 turnspeed from `HeliFlyCircle`.

Split from #15803.

Will get a follow-up later, replacing `LandWhenIdle` with a more flexible `ActionsWhenIdle`.

reaperrr committed:
Add IdleTurnSpeed to Aircraft

Instead of hardcoding 1/3 of normal TurnSpeed on HeliFlyCircle.

reaperrr committed:
Use INotifyIdle on Aircraft

For now only to trigger landing or circling.

reaperrr committed:
Remove queueing HeliFlyCircle from AutoCarryall

pchote commented on OpenRA.Mods.Common/Traits/Air/Aircraft.cs:
```diff
@@ -502,6 +506,29 @@ public void ModifyDeathActorInit(Actor self, TypeDictionary init)
+			OnBecomingIdle(self);
+		}
+
+		protected virtual void OnBecomingIdle(Actor self)
```
This introduces a one-tick stall where aircraft stop moving when transitioning from flying to circling.

A specific workaround that may work here is to RunActivity at the end, but I suspect a proper fix needs to be implemented deeper in the `INotifyBecomingIdle` code.

reaperrr replied:
Fixed it in `Actor` via `RunActivity`, I couldn't come up with anything else.

reaperrr committed:
Fix empty activity tick when becoming idle

Activities usually don't do much more than cleanup on their last, 'null' tick.
That, combined with queued activities normally only starting to tick on the next tick,
would lead to visible 1-tick 'gaps' between movement activities.
Non-movement activities would suffer from the same problem,
only with different (presumably less noticable) symptoms.

Now we start ticking any activity that was queued from OnBecomingIdle
immediately, to avoid that issue.

TheChosenEvilOne reviewed (dismissed):
LGTM :+1:

pchote reviewed (approved):
Force-pushed with a rebase.

pchote commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/38d079f381e3de60fe324ddc43d00e651113ef2e
